src/cfn/index.ts

```diff
@@ -84,13 +85,40 @@ export type StackArgs = {
   CommandsBefore?: string[]
 }
 
+const stackArgsProperties: Array<keyof StackArgs> = [
+  'ApprovedTemplateLocation',
+  'AssumeRoleARN',
+  'Capabilities',
+  'ClientRequestToken',
+  'CommandsBefore',
+  'DisableRollback',
+  'EnableTerminationProtection',
+  'NotificationARNs',
+  'OnFailure',
+  'Parameters',
+  'Profile',
+  'Region',
+  'ResourceTypes',
+  'RoleARN',
+  'ServiceRoleARN',
+  'StackName',
+  'StackPolicy',
+  'Tags',
+  'Template',
+  'TimeoutInMinutes',
+  'UsePreviousTemplate',
+];
+
+const stackArgsMetaProperties = ['$imports', '$defs'];
```

Comment:
What about `$merge` if someone were to use that at the top level?

Reply:
I thought about that and I agree that it should be supported – I debated moving the warnings check until after `transform` is called, but mistyping something like `AssumeRoleARN` or `Profile` could result in fetching imports to fail and it would not be obvious that the parameter name is the culprit. 

I think that even though it is less helpful, it probably makes more sense to move it until after `transform`.